WireDomain/Sources/WireDomain/UseCases/IsBuildBlacklistedUseCase.swift

```diff
@@ -0,0 +1,65 @@
+//
+// Wire
+// Copyright (C) 2025 Wire Swiss GmbH
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with this program. If not, see http://www.gnu.org/licenses/.
+//
+
+import Foundation
+import WireNetwork
+
+public protocol IsBuildBlacklistedUseCase {
+
+    func invoke() async throws -> Bool
+
+}
+
+public enum IsBuildBlacklistedUseCaseError: Error {
+
+    case decodingFailed(message: String)
+
+}
+
+public struct IsBuildBlacklistedUseCaseImpl: IsBuildBlacklistedUseCase {
+
+    let currentBuildNumber: String
+    let api: any BlacklistAPI
+
+    public init(
+        currentBuildNumber: String,
+        api: any BlacklistAPI
+    ) {
+        self.currentBuildNumber = currentBuildNumber
+        self.api = api
+    }
+
+    public func invoke() async throws -> Bool {
+        let blacklist = try await api.getBlacklist()
```

Comment:
question: we always do the network request here, so what happens if :
1 run the app donwload the blacklist and shows the popup
2 turn on airplane mode
3 run the app again
Do i still get stop because of the blacklist?